xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
 {
-    if (s.length() != 3)
-	throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
-    return new TfIdfWeight(s);
+    if (param_s > 0 || param_delta > 0) {
```

Comment:
If I have a serialised `TfIdfWeight`, _without_ knowing whether it's using a normalisation string or the params (which, see previous comment about constructors, I don't think is the distinction you want), how do I know how to unserialise it? A serialised version needs to be able to be serialised without knowing how it was built.